services/datamanager/builtin/builtin_test.go

```diff
@@ -957,3 +906,123 @@ func getTestModelManagerConstructor(t *testing.T, server rpc.Server, zipFileName
 		return model.NewManager(logger, cfg.Cloud.ID, client, conn, &mockClient{zipFileName: zipFileName})
 	}
 }
+
+func TestDataCapture(t *testing.T) {
+	tests := []struct {
+		name                  string
+		initialDisabledStatus bool
+		newDisabledStatus     bool
+	}{
+		{
+			"Config with data capture disabled should capture nothing.",
+			true,
+			true,
+		},
+		{
+			"Config with data capture enabled should capture data.",
+			false,
+			false,
+		},
+		{
+			"Disabling data capture should cause it to stop.",
+			false,
+			true,
+		},
+		{
+			"Enabling data capture should cause it to begin.",
+			true,
+			false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			tmpDir, err := ioutil.TempDir("", "")
+			defer os.RemoveAll(tmpDir)
+			test.That(t, err, test.ShouldBeNil)
+
+			rpcServer, _ := buildAndStartLocalServer(t)
+			defer func() {
+				err := rpcServer.Stop()
+				test.That(t, err, test.ShouldBeNil)
+			}()
+			dmsvc := newTestDataManager(t, "arm1", "")
+			dmsvc.SetSyncerConstructor(getTestSyncerConstructor(t, rpcServer))
+			testCfg := setupConfig(t, configPath)
+			svcConfig, ok, err := getServiceConfig(testCfg)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, ok, test.ShouldBeTrue)
+
+			svcConfig.CaptureDisabled = tc.initialDisabledStatus
+			svcConfig.ScheduledSyncDisabled = true
+			svcConfig.CaptureDir = tmpDir
+			err = dmsvc.Update(context.Background(), testCfg)
+			test.That(t, err, test.ShouldBeNil)
+
+			// let run for a second
```

Comment:
supernit: technically 25 milliseconds, so something like "Run capture for a moment"